### DISCUSSION TITLE
New Features about Chaos Experiment Observability

### DISCUSSION THREAD

**Discussion by STRRL** [chaos-mesh/chaos-mesh#3012] [Ideas]
Hi folks! I am very excited to share some new ideas with Chaos Experiment Observability.

There are 2 basic ideas about that:

- Enrich the observability of chaos experiment itself
- Improve the user experience about exploring the observability of affected applications

## Enrich the observability of chaos experiment itself

We are often asked by users: how to find out that the chaos experiments are really "worked"? We would provide different answers to different situations, for example:

- use `ping` or `netstat` for observing NetworkChaos
- use `top` for observing Stresschaos
- use `ls`, `stat` or `fio` for observing IOChaos
- ...

How about we directly providing the metrics with affected "things" the chaos experiment?

For example:

- For StressChaos, when we looking at the chaos experiment on chaos dashboard, display the "real-time" cpu and memory usage line-charts.
- For IOChaos, show the table of all the effected files the how much read/write operations on it.

And the metrics would also be exported as prometheus `/health` endpoint, it could use it with your existed monitoring stack.

But one of the unconsidered things is:

- I am not sure this feature would keep useful as common-used features, or the user are only interested it when "exploring Chaos Mesh" as a fresh user.  

## Improve the user experience about exploring the observability of affected applications

We want to integrate some monitoring stack within Chaos Dashboard. For example, prometheus, grafana, Elastic Stack and Datalog.

When user using Chaos Mesh to inject some chaos experiment into an application, there **must be** some "core metrics" is important. In other words, the "core metrics" to measure the "steady state".

I just wondering "how about show that core metrics line-charts just close to the experiment on Chaos Dashboard?"

It could might reduce the steps for manually measure the steady state when injecting chaos experiment.

> We have provided the grafana-plugin to display the chaos experiments as grafana annotations. It's another way to do that.

And I have drafted an RFC about the second idea: https://github.com/chaos-mesh/rfcs/pull/38, but it seems it is not good-enough idea now.

How do you think about these ideas? Or give us any other ideas!

Feel free to leave any comments, it's very helpful to us!

**Comment by xlgao-zju** [comment]
this is so cool, I am looking forward to this feature.

**Comment by g1eny0ung** [comment]
If you come from the LFX program, please read https://cloud-native.slack.com/archives/C0193VAV272/p1706923006128609 first, thanks.
